### PR TITLE
Updates for VHBB

### PR DESCRIPTION
**Pull request by DylanHsu**
benedikt should sign off on this

### CODE REVIEW

**Comment by DylanHsu** [comment]
Renamed this PR. Testing now

**Comment by DylanHsu** [comment]
Test looks ok, please review and merge

Testing subjet medium CSV b-tag SFs:
http://t3serv001.mit.edu/~dhsu/PandaAnalysis_on_8_0_29/Flat/test/sf_sjbtag0_009_wlnhbb.pdf
http://t3serv001.mit.edu/~dhsu/PandaAnalysis_on_8_0_29/Flat/test/sf_sjbtag0BUp_009_wlnhbb.pdf
http://t3serv001.mit.edu/~dhsu/PandaAnalysis_on_8_0_29/Flat/test/sf_sjbtag0MUp_009_wlnhbb.pdf
http://t3serv001.mit.edu/~dhsu/PandaAnalysis_on_8_0_29/Flat/test/sf_sjbtag2_009_wlnhbb.pdf
http://t3serv001.mit.edu/~dhsu/PandaAnalysis_on_8_0_29/Flat/test/sf_sjbtag2BUp_009_wlnhbb.pdf
http://t3serv001.mit.edu/~dhsu/PandaAnalysis_on_8_0_29/Flat/test/sf_sjbtag2MUp_009_wlnhbb.pdf

**Comment by DylanHsu** [comment]
new variables "nBGenJets" seems a little screwed up... or is it just me?
Running on WHbb signal file:
```
root [3] events->Scan("nB:nBGenJets","hbbpt>100")
************************************
*    Row   *        nB * nBGenJets *
************************************
*        3 *         2 *        11 *
*        4 *         2 *         2 *
*        7 *         2 *         7 *
*       12 *         2 *        12 *
*       13 *         2 *         3 *
*       16 *         2 *         6 *
*       31 *         2 *         9 *
*       33 *         2 *         9 *
*       41 *         2 *        13 *
*       42 *         2 *         3 *
*       48 *         4 *         9 *
*       50 *         2 *         3 *
*       51 *         2 *         7 *
*       53 *         3 *         9 *
*       56 *         2 *         5 *
*       59 *         2 *         5 *
*       60 *         2 *        11 *
*       62 *         2 *         4 *
*       80 *         2 *         3 *
*       82 *         2 *         7 *
*       86 *         2 *         3 *
*       93 *         2 *         5 *
*       95 *         2 *         3 *
*       97 *         2 *         3 *
*      109 *         2 *         0 *
Type <CR> to continue or q to quit ==> 
*      120 *         2 *         3 *
*      126 *         2 *         7 *
*      132 *         2 *         7 *
*      135 *         2 *         8 *
*      139 *         2 *         5 *
*      141 *         2 *        13 *
*      145 *         2 *        13 *
*      146 *         2 *         5 *
*      148 *         2 *         6 *
*      155 *         2 *         6 *
*      157 *         2 *         3 *
*      159 *         2 *         3 *
*      162 *         2 *         8 *
*      164 *         2 *         3 *
*      165 *         2 *         5 *
*      171 *         2 *         3 *
*      172 *         2 *         7 *
*      175 *         2 *         9 *
*      185 *         2 *         9 *
*      187 *         2 *        11 *
*      188 *         2 *         4 *
*      197 *         2 *         3 *
*      207 *         2 *         7 *
*      211 *         2 *         8 *
*      215 *         2 *         4 *
```

File: Flat/src/GeneralTree.cc [hunk 1]
```diff
@@ -876,6 +877,7 @@ void GeneralTree::WriteTree(TTree *t) {
   if (fatjet || hfCounting) {
```
Comment (sidnarayanan) [inline Flat/src/GeneralTree.cc]: Make this `if (hfCounting) { ...`

File: Flat/src/ModulesFatJets.cc [hunk 2]
```diff
@@ -758,12 +772,12 @@ void PandaAnalyzer::FatjetMatching()
         }
 
       } // loop over subjets
-
-      EvalBTagSF(sj_btagcands,sj_sf_cent,GeneralTree::bCent,GeneralTree::bSubJet);
-      EvalBTagSF(sj_btagcands,sj_sf_bUp,GeneralTree::bBUp,GeneralTree::bSubJet);
-      EvalBTagSF(sj_btagcands,sj_sf_bDown,GeneralTree::bBDown,GeneralTree::bSubJet);
-      EvalBTagSF(sj_btagcands,sj_sf_mUp,GeneralTree::bMUp,GeneralTree::bSubJet);
-      EvalBTagSF(sj_btagcands,sj_sf_mDown,GeneralTree::bMDown,GeneralTree::bSubJet);
+      bool do2=true;
```
Comment (sidnarayanan) [inline Flat/src/ModulesFatJets.cc]: What's this?

File: Flat/src/ModulesLepPho.cc [hunk 3]
```diff
@@ -131,8 +131,11 @@ void PandaAnalyzer::ComplicatedLeptons() {
   looseLep1PdgId=-1, looseLep2PdgId=-1, looseLep3PdgId=-1, looseLep4PdgId=-1;
   for (auto& ele : event.electrons) {
     float pt = ele.smearedPt; float eta = ele.eta(); float aeta = fabs(eta);
-    if (pt<10 || aeta>2.5 /* || (aeta>1.4442 && aeta<1.566) */) continue;
-    if (!ele.veto) continue;
+    if(analysis->hbb) {
```
Comment (sidnarayanan) [inline Flat/src/ModulesLepPho.cc]: `if (`

File: Flat/src/ModulesFatJets.cc (hunk 2 above)
Comment (DylanHsu) [reply Flat/src/ModulesFatJets.cc]: it's local and used on the very next line, trying to keep the code self explanatory

File: Flat/src/ModulesLepPho.cc [hunk 4]
```diff
@@ -210,9 +213,12 @@ void PandaAnalyzer::ComplicatedLeptons() {
         ptCorrection=rochesterCorrection->kScaleAndSmearMC((int)mu.charge, pt, eta, mu.phi(), mu.trkLayersWithMmt, random1, random2, 0, 0);
       }
       pt *= ptCorrection;
-    } 
-    if (pt<10 || aeta>2.4) continue;
-    if (!mu.loose) continue;
+    }
+    if(analysis->hbb) {
```
Comment (sidnarayanan) [inline Flat/src/ModulesLepPho.cc]: ditto

File: Flat/src/ModulesFatJets.cc (hunk 2 above)
Comment (sidnarayanan) [reply Flat/src/ModulesFatJets.cc]: It's not used on the next line.

Comment (sidnarayanan) [reply Flat/src/ModulesFatJets.cc]: Actually I'd prefer we just pass literals. The other 4 arguments of the function aren't exactly self-explanatory.

Comment (DylanHsu) [reply Flat/src/ModulesFatJets.cc]: ok

**Comment by DylanHsu** [comment]
fixed the things sid already asked for , testing over 10k events again

**Comment by DylanHsu** [comment]
Looks a bit more reasonable now for a WH(bb) signal file, but I'd argue in favor of using nB after seeing these values.
```
root [1] events->Scan("nB:nBGenJets","hbbpt>100")
************************************
*    Row   *        nB * nBGenJets *
************************************
*        3 *         2 *         2 *
*        4 *         2 *         1 *
*        7 *         2 *         2 *
*       12 *         2 *         2 *
*       13 *         2 *         2 *
*       16 *         2 *         1 *
*       31 *         2 *         2 *
*       33 *         2 *         2 *
*       41 *         2 *         2 *
*       42 *         2 *         2 *
*       48 *         4 *         2 *
*       50 *         2 *         2 *
*       51 *         2 *         2 *
*       53 *         3 *         2 *
*       56 *         2 *         1 *
*       59 *         2 *         2 *
*       60 *         2 *         2 *
*       62 *         2 *         1 *
*       80 *         2 *         2 *
*       82 *         2 *         2 *
*       86 *         2 *         1 *
*       93 *         2 *         2 *
*       95 *         2 *         2 *
*       97 *         2 *         2 *
*      109 *         2 *         0 *
Type <CR> to continue or q to quit ==> 
*      120 *         2 *         2 *
*      126 *         2 *         2 *
*      132 *         2 *         2 *
*      135 *         2 *         1 *
*      139 *         2 *         2 *
*      141 *         2 *         2 *
*      145 *         2 *         2 *
*      146 *         2 *         2 *
*      148 *         2 *         1 *
*      155 *         2 *         1 *
*      157 *         2 *         1 *
*      159 *         2 *         1 *
*      162 *         2 *         2 *
*      164 *         2 *         2 *
*      165 *         2 *         2 *
*      171 *         2 *         2 *
*      172 *         2 *         2 *
*      175 *         2 *         2 *
*      185 *         2 *         2 *
*      187 *         2 *         2 *
*      188 *         2 *         1 *
*      197 *         2 *         2 *
*      207 *         2 *         2 *
*      211 *         2 *         1 *
*      215 *         2 *         1 *
```

**Pull request merged**